### PR TITLE
Added iSend SMS Provider

### PR DESCRIPTION
**Pull request by Saff-Elli-Khan**
### What change does this PR introduce?

Added a new SMS provider called isend-sms. isend is a Libyan SMS provider company which provides services just in Libya and is cheaper compared to Twilio or any other provider, for Libyan developers.

### Why was this change needed?

Support for a Libyan SMS provider company isend-sms https://send.com.ly/

### CODE REVIEW

File: packages/application-generic/package.json [hunk 1]
```diff
@@ -55,6 +55,7 @@
     "@novu/bandwidth": "^0.22.0",
     "@novu/braze": "^0.22.0",
     "@novu/bulk-sms": "^0.22.0",
+    "@novu/isend-sms": "^0.16.3",
```
Comment (Cliftonz) [inline packages/application-generic/package.json:58]: This should be the same as the other providers

File: providers/isend-sms/package.json [hunk 2]
```diff
@@ -0,0 +1,78 @@
+{
+  "name": "@novu/isend-sms",
+  "version": "0.16.3",
```
Comment (Cliftonz) [inline providers/isend-sms/package.json:3]: This should be updated to the current version

**Review by Cliftonz — CHANGES_REQUESTED** [review]

**Comment by Cliftonz** [comment]
@Saff-Elli-Khan The CI is failing on a unit test and there is a missing cspell value

**Comment by Saff-Elli-Khan** [comment]
@Cliftonz I have fixed the spelling test, but I have no idea why other tests are failing in the CI, I added the provider exactly according to the docs provided by Novu. It is really confusing to me can you please point out where is the problem?

File: providers/isend-sms/jest.config.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+/** @type {import('ts-jest/dist/types').InitialOptionsTsJest} */
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'node',
```
Comment (rifont) [inline providers/isend-sms/jest.config.js:4]: To resolve the Jest test issue [here](https://github.com/novuhq/novu/actions/runs/7260977977/job/19781519293?pr=5000#step:5:71), a module name mapper is required due to Jest's lack of support for ES modules.

An example in another provider is available [here](https://github.com/novuhq/novu/blob/1950a191ffb351875c7d79a16237ed9e23000226/providers/resend/jest.config.js#L6).
```suggestion
  testEnvironment: 'node',
  moduleNameMapper: {
    axios: 'axios/dist/node/axios.cjs',
  },
```

**Comment by Saff-Elli-Khan** [comment]
@Cliftonz @rifont the test is still failing for some reason! Can you please help? I really need this provider on production asap!

**Review by Cliftonz — APPROVED** [review]

**Comment by Cliftonz** [comment]
@Saff-Elli-Khan this provider should be out in the next release around the beginning of February

**Comment by Saff-Elli-Khan** [comment]
> @Saff-Elli-Khan this provider should be out in the next release around the beginning of February

Great! Thanks :)

**Pull request merged**